internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -52,11 +66,63 @@ func (c *CodePipeline) GetPipeline(name string) (*Pipeline, error) {
 	if err != nil {
 		return nil, fmt.Errorf("get pipeline %s: %w", name, err)
 	}
-	pipeline := &Pipeline{
-		Name: aws.StringValue(resp.Pipeline.Name),
+
+	pipeline := resp.Pipeline
+	metadata := resp.Metadata
+	arn := aws.StringValue(metadata.PipelineArn)
+	region, err := c.getPipelineRegion(arn)
+	if err != nil {
+		return nil, err
+	}
+	accountId, err := c.getPipelineAccountId(arn)
+	if err != nil {
+		return nil, err
+	}
+
+	var stages []Stage
+	for _, s := range pipeline.Stages {
+		name := aws.StringValue(s.Name)
+		action := s.Actions[0]
+
+		category := aws.StringValue(action.ActionTypeId.Category)
+		provider := aws.StringValue(action.ActionTypeId.Provider)
+		config := action.Configuration
+
+		var details string
+		if category == "Source" {
+			details = fmt.Sprintf("Repository: %s/%s", aws.StringValue(config["Owner"]), aws.StringValue(config["Repo"]))
+		}
+		if category == "Build" {
+			details = fmt.Sprintf("BuildProject: %s", aws.StringValue(config["ProjectName"]))
+		}
+		if category == "Deploy" {
+			details = fmt.Sprintf("StackName: %s", aws.StringValue(config["StackName"]))
+		}
+
+		stage := Stage{
+			Name:     name,
+			Category: category,
+			Provider: provider,
+			Details:  details,
+		}
+		stages = append(stages, stage)
 	}
 
-	return pipeline, nil
+	return &Pipeline{
+		Name:      aws.StringValue(pipeline.Name),
+		Region:    region,
+		AccountID: accountId,
+		Stages:    stages,
+		CreatedAt: metadata.Created,
+		UpdatedAt: metadata.Updated,
+	}, nil
+}
+
+// HumanString returns the stringified Stage struct with human readable format.
+// Example output:
+//   DeployTo-test	Deploy	Cloudformation	stackname: dinder-test-test
+func (s *Stage) HumanString() string {
+	return fmt.Sprintf("  %s\t%s\t%s\t%s\n", s.Name, s.Category, s.Provider, s.Details)
```

Comment:
Hm, since the prefixed whitepace is kind of part of the point of extracting HumanString, I think it'd be easier to keep it this way for now until this needs to be reused, as it was following the example of  https://github.com/aws/amazon-ecs-cli-v2/blob/master/internal/pkg/aws/ecs/ecs.go#L69 and the like.